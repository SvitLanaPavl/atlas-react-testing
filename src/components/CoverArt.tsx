import React from 'react'
import placeholder from '../assets/placeholder.svg'

const CoverArt: React.FC = () => {
  return (
    <div className='relative mb-6 aspect-square'>
      <img src={placeholder} alt='placeholder image' className='h-full w-full rounded-md aspect-[400/400] object-cover block align-middle' />
    </div>
  )
}

export default CoverArt