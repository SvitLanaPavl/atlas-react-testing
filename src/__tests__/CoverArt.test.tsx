import { render } from '@testing-library/react';
import { expect, test } from 'vitest';
import CoverArt from '@/components/CoverArt';

test('CoverArt renders correctly with valid image', () => {
  const { container } = render(<CoverArt cover='https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn4.webp' />);
  expect(container).toMatchSnapshot();
});