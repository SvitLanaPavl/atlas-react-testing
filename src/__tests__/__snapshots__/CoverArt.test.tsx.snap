// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CoverArt renders correctly with valid image 1`] = `
<div>
  <div
    class="relative mb-6 aspect-square"
  >
    <img
      alt="placeholder image"
      class="h-full w-full rounded-md aspect-[400/400] object-cover block align-middle"
      src="https://raw.githubusercontent.com/atlas-jswank/atlas-music-player-api/main/images/albumn4.webp"
    />
  </div>
</div>
`;
